default.nix

```diff
@@ -0,0 +1,61 @@
+let
+  localLib = import ./lib.nix;
+  jemallocOverlay = self: super: {
+    # jemalloc has a bug that caused cardano-sl-db to fail to link (via
+    # rocksdb, which can use jemalloc).
+    # https://github.com/jemalloc/jemalloc/issues/937
+    # Using jemalloc 510 with the --disable-initial-exec-tls flag seems to
+    # fix it.
+    jemalloc = self.callPackage ./nix/jemalloc/jemalloc510.nix {};
+  };
+in
+{ system ? builtins.currentSystem
+, pkgs ? (import (localLib.fetchNixPkgs) { inherit system config; overlays = [ jemallocOverlay ]; })
+, config ? {}
+}:
+
+with pkgs.lib;
+with pkgs.haskell.lib;
+
+let
+  plutusPkgs = ((import ./pkgs { inherit pkgs; }).override {
+    ghc = overrideDerivation pkgs.haskell.compiler.ghc822 (drv: {
+      patches = drv.patches ++ [ ./ghc-8.0.2-darwin-rec-link.patch ];
+    });
+  });
+  cleanSourceFilter = with pkgs.stdenv;
+    name: type: let baseName = baseNameOf (toString name); in ! (
+      # Filter out .git repo
+      (type == "directory" && baseName == ".git") ||
+      # Filter out editor backup / swap files.
+      lib.hasSuffix "~" baseName ||
+      builtins.match "^\\.sw[a-z]$" baseName != null ||
+      builtins.match "^\\..*\\.sw[a-z]$" baseName != null ||
+
+      # Filter out locally generated/downloaded things.
+      baseName == "dist" ||
+
+      # Filter out the files which I'm editing often.
+      lib.hasSuffix ".nix" baseName ||
+      lib.hasSuffix ".dhall" baseName ||
+      # Filter out nix-build result symlinks
+      (type == "symlink" && lib.hasPrefix "result" baseName) ||
+      (type == "directory" && baseName == ".stack-work")
+    );
+  source = builtins.filterSource cleanSourceFilter ./.;
```

Comment:
this filter is unused

Reply:
ah, I stopped using it as a troubleshooting step with the hydra issue we were debugging, back in.